src/qibo/models/circuit.py

```diff
@@ -512,7 +580,7 @@ def with_pauli_noise(self, noise_map: NoiseMapType):
                 noisy_circuit.add(noise_gate)
         return noisy_circuit
 
-    def add(self, gate):
+    def add(self, gate, position=-1):
         """Add a gate to a given queue.
 
         Args:
```

Comment:
In case we decide to keep this, the docstrings must be updated by adding the new `position` argument.